circle.yml

```diff
@@ -20,6 +25,52 @@ jobs:
           command: npm test -- --reporters=default --reporters=jest-junit --ci --runInBand
       - store_test_results:
           path: .
+
+  build-frontend:
+    docker:
+      - image: circleci/node:lts
+    working_directory: ~/project/frontend
+    steps:
+      - checkout:
+          path: ~/project
+      - restore_cache:
+          key: dependencies-{{ checksum "package-lock.json" }}
+      - run:
+          name: Installing frontend dependencies
+          command: npm install
+      - save_cache:
+          paths:
+            - node_modules
+          key: dependencies-{{ checksum "package-lock.json" }}
+      - run:
+          name: Build React frontend
+          command: npm run build
```

Comment:
Don't we need to generate any `.env` file for the frontend deployment?

Reply:
No, the environment variables are taken directly from the CircleCI configuration and used in the build.
You can see the variables [here](https://circleci.com/gh/open-craft/opencraft/edit#env-vars), prefixed with `FRONTEND_STAGE_`.

Reply:
Oh, nice! Just curious - how does it work? Do we have some custom build scripts for supporting this naming convention?

Reply:
Ah, I see what you mean. Yeah, I'm not passing the variables correctly to the react app, just for the AWS deployment script. 
Let me copy paste a few lines from the sprints app and fix this up :stuck_out_tongue:

Reply:
Haha, sure :P